data/transactions/logic/doc.go

```diff
@@ -287,6 +289,7 @@ var opDocExtras = map[string]string{
 	"itxn_field":          "`itxn_field` fails if X is of the wrong type for F, including a byte array of the wrong size for use as an address when F is an address field. `itxn_field` also fails if X is an account or asset that does not appear in `txn.Accounts` or `txn.ForeignAssets` of the top-level transaction. (Setting addresses in asset creation are exempted from this requirement.)",
 	"itxn_submit":         "`itxn_submit` resets the current transaction so that it can not be resubmitted. A new `itxn_begin` is required to prepare another inner transaction.",
 	"base64_decode":       "Decodes X using the base64 encoding E. Specify the encoding with an immediate arg either as URL and Filename Safe (`URLEncoding`) or Standard (`StdEncoding`). See <a href=\"https://rfc-editor.org/rfc/rfc4648.html#section-4\">RFC 4648</a> (sections 4 and 5). It is assumed that the encoding ends with the exact number of `=` padding characters as required by the RFC. When padding occurs, any unused pad bits in the encoding must be set to zero or the decoding will fail. The special cases of `\\n` and `\\r` are allowed but completely ignored. An error will result when attempting to decode a string with a character that is not in the encoding alphabet or not one of `=`, `\\r`, or `\\n`.",
+	"json_ref":            "specify the return type with an immediate arg either as JSONInt or JSONString. When B is a nested key of json object A, A.B returns the value of B. Providing a json with keys containing . will return incorrect value. See <a href=\"https://datatracker.ietf.org/doc/html/rfc7159.html\">RFC 7159</a> for additional json specifications",
```

Comment:
Since we are _really_ returning bytes and not characters (and reading bytes as characters), we need to be clear about any encoding that's going on.  What does JSON say about encodings? Maybe it only allows low-ascii?  That'd be nice.  Otherwise we need to talking more about how we convert the bytes to characters (utf-8?) and then how we convert the string extracted to bytes.  Again, utf-8?

Reply:
I did some reading and it seems to assume the bytes going into unmashalling are utf-8.  That's good, and well defined.  And I think it's fine for use to encode to bytes as utf-8 as well.  The "BOM" needs some attention though - it seems like it is allowed but ignored in some cases?